### ISSUE TITLE
Remove requirement for an Adapter on MarkerView

### ISSUE THREAD

**Issue by tobrun** [mapbox/mapbox-gl-native#5214]
Currently we require a developer to provide an `MarkerViewAdapter` to be able show a `MarkerView`. 
Current game plan is to remove this requirement and handle the logic for them. Internally the base `MarkerView` will be backed up by `ImageMarkerViewAdapter` and will leverage the `Icon` attribute in Marker.

**Comment by Schumi09** [comment]
Sounds amazing, this will make it much more handy! I couldn't even convert the icon's bitmap of the passed marker back to a drawable and use it for `viewHolder.icon.setImageDrawable())`
Guess I don't have to write my own View classes anymore just to pass a drawable directly.

**Comment by tobrun** [comment]
@Schumi09

>  I couldn't even convert the icon's bitmap of the passed marker back to a drawable and use it for viewHolder.icon.setImageDrawable())

With the changes made you can achieve that as well ;)

**Comment by tobrun** [comment]
This is the code a developer needs to write to show a normal MarkerView:

```java
   MarkerViewOptions options = new MarkerViewOptions();
                options.title("United States");
                options.icon(IconFactory.getInstance(mMapView.getContext())
                        .fromResource(R.drawable.ic_us));
                options.position(new LatLng(38.899774, -77.023237));
                options.flat(true);
    markerView = mapboxMap.addMarker(options);
```

Below you can see where I'm updating the Icon every 3 seconds through `markerView.setIcon()`:

![ezgif com-video-to-gif](https://cloud.githubusercontent.com/assets/2151639/15741837/c7446766-28bb-11e6-9ab3-4fa43a85db22.gif)

**Comment by Schumi09** [comment]
Good job :+1: 

Will it make it into the tomorrow's snapshot? 

Maybe you could also test the behaviour of the issues I have experienced concerning the removing and adding new markerviews (#5161). Moreover, you said that the View things will provide a fix to disappearing Icons/Markers during runtime. Did it fix it?

**Comment by tobrun** [comment]
> Maybe you could also test the behaviour of the issues I have experienced concerning the removing and adding new markerviews (#5161). 
 
I'm planning to do some general improvement to this system in #5076, this includes looking into above.

> you said that the View things will provide a fix to disappearing Icons/Markers during runtime. Did it fix it?

Yes this issue should be resolved by using this API.

**Comment by tobrun** [comment]
merged with 087cad0, closing
